<a href="https://colab.research.google.com/github/GuilleLondero/challenge_one_DS_telecomX/blob/main/ChallengerTelecomX.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**📌 Extracción**

In [185]:
# 1. Importar las bibliotecas necesarias
import pandas as pd
import requests
import json

# 2. URL de la API de TelecomX
url = "https://raw.githubusercontent.com/ingridcristh/challenge2-data-science-LATAM/refs/heads/main/TelecomX_Data.json"

# 3. Realizo la petición HTTP para obtener los datos
try:
  response = requests.get(url)  # Obtengo los Datos mediante el metodo HTPP get
  response.raise_for_status()   # Lanza excepción si hay error HTTP
  data = response.json()        # Convierto la respuesta JSON a un diccionario de Python

  df = pd.DataFrame(data)       # Convierto los datos a DataFrame de Pandas

except requests.exceptions.RequestException as e:
    print(f" Error al descargar los datos: {e}")
except json.JSONDecodeError as e:
    print(f" Error al procesar JSON: {e}")
except Exception as e:
    print(f" Error inesperado: {e}")

# 4. Muestro el data frame obtenido
df

customerID Churn                                           customer  \
0     0002-ORFBO    No  {'gender': 'Female', 'SeniorCitizen': 0, 'Part...   
1     0003-MKNFE    No  {'gender': 'Male', 'SeniorCitizen': 0, 'Partne...   
2     0004-TLHLJ   Yes  {'gender': 'Male', 'SeniorCitizen': 0, 'Partne...   
3     0011-IGKFF   Yes  {'gender': 'Male', 'SeniorCitizen': 1, 'Partne...   
4     0013-EXCHZ   Yes  {'gender': 'Female', 'SeniorCitizen': 1, 'Part...   
...          ...   ...                                                ...   
7262  9987-LUTYD    No  {'gender': 'Female', 'SeniorCitizen': 0, 'Part...   
7263  9992-RRAMN   Yes  {'gender': 'Male', 'SeniorCitizen': 0, 'Partne...   
7264  9992-UJOEL    No  {'gender': 'Male', 'SeniorCitizen': 0, 'Partne...   
7265  9993-LHIEB    No  {'gender': 'Male', 'SeniorCitizen': 0, 'Partne...   
7266  9995-HOTOH    No  {'gender': 'Male', 'SeniorCitizen': 0, 'Partne...   

                                                  phone  \
0        {'PhoneService': 'Yes', 'MultipleLines': 'No'}   
1       {'PhoneService': 'Yes', 'MultipleLines': 'Yes'}   
2        {'PhoneService': 'Yes', 'MultipleLines': 'No'}   
3        {'PhoneService': 'Yes', 'MultipleLines': 'No'}   
4        {'PhoneService': 'Yes', 'MultipleLines': 'No'}   
...                                                 ...   
7262     {'PhoneService': 'Yes', 'MultipleLines': 'No'}   
7263    {'PhoneService': 'Yes', 'MultipleLines': 'Yes'}   
7264     {'PhoneService': 'Yes', 'MultipleLines': 'No'}   
7265     {'PhoneService': 'Yes', 'MultipleLines': 'No'}   
7266  {'PhoneService': 'No', 'MultipleLines': 'No ph...   

                                               internet  \
0     {'InternetService': 'DSL', 'OnlineSecurity': '...   
1     {'InternetService': 'DSL', 'OnlineSecurity': '...   
2     {'InternetService': 'Fiber optic', 'OnlineSecu...   
3     {'InternetService': 'Fiber optic', 'OnlineSecu...   
4     {'InternetService': 'Fiber optic', 'OnlineSecu...   
...                                                 ...   
7262  {'InternetService': 'DSL', 'OnlineSecurity': '...   
7263  {'InternetService': 'Fiber optic', 'OnlineSecu...   
7264  {'InternetService': 'DSL', 'OnlineSecurity': '...   
7265  {'InternetService': 'DSL', 'OnlineSecurity': '...   
7266  {'InternetService': 'DSL', 'OnlineSecurity': '...   

                                                account  
0     {'Contract': 'One year', 'PaperlessBilling': '...  
1     {'Contract': 'Month-to-month', 'PaperlessBilli...  
2     {'Contract': 'Month-to-month', 'PaperlessBilli...  
3     {'Contract': 'Month-to-month', 'PaperlessBilli...  
4     {'Contract': 'Month-to-month', 'PaperlessBilli...  
...                                                 ...  
7262  {'Contract': 'One year', 'PaperlessBilling': '...  
7263  {'Contract': 'Month-to-month', 'PaperlessBilli...  
7264  {'Contract': 'Month-to-month', 'PaperlessBilli...  
7265  {'Contract': 'Two year', 'PaperlessBilling': '...  
7266  {'Contract': 'Two year', 'PaperlessBilling': '...  

[7267 rows x 6 columns]

## **Normalizando el Data Frame:**

In [186]:
df_normalize = pd.json_normalize(data)
df_normalize

customerID Churn customer.gender  customer.SeniorCitizen  \
0     0002-ORFBO    No          Female                       0   
1     0003-MKNFE    No            Male                       0   
2     0004-TLHLJ   Yes            Male                       0   
3     0011-IGKFF   Yes            Male                       1   
4     0013-EXCHZ   Yes          Female                       1   
...          ...   ...             ...                     ...   
7262  9987-LUTYD    No          Female                       0   
7263  9992-RRAMN   Yes            Male                       0   
7264  9992-UJOEL    No            Male                       0   
7265  9993-LHIEB    No            Male                       0   
7266  9995-HOTOH    No            Male                       0   

     customer.Partner customer.Dependents  customer.tenure phone.PhoneService  \
0                 Yes                 Yes                9                Yes   
1                  No                  No                9                Yes   
2                  No                  No                4                Yes   
3                 Yes                  No               13                Yes   
4                 Yes                  No                3                Yes   
...               ...                 ...              ...                ...   
7262               No                  No               13                Yes   
7263              Yes                  No               22                Yes   
7264               No                  No                2                Yes   
7265              Yes                 Yes               67                Yes   
7266              Yes                 Yes               63                 No   

     phone.MultipleLines internet.InternetService  ... internet.OnlineBackup  \
0                     No                      DSL  ...                   Yes   
1                    Yes                      DSL  ...                    No   
2                     No              Fiber optic  ...                    No   
3                     No              Fiber optic  ...                   Yes   
4                     No              Fiber optic  ...                    No   
...                  ...                      ...  ...                   ...   
7262                  No                      DSL  ...                    No   
7263                 Yes              Fiber optic  ...                    No   
7264                  No                      DSL  ...                   Yes   
7265                  No                      DSL  ...                    No   
7266    No phone service                      DSL  ...                   Yes   

     internet.DeviceProtection internet.TechSupport internet.StreamingTV  \
0                           No                  Yes                  Yes   
1                           No                   No                   No   
2                          Yes                   No                   No   
3                          Yes                   No                  Yes   
4                           No                  Yes                  Yes   
...                        ...                  ...                  ...   
7262                        No                  Yes                   No   
7263                        No                   No                   No   
7264                        No                   No                   No   
7265                       Yes                  Yes                   No   
7266                       Yes                   No                  Yes   

     internet.StreamingMovies account.Contract account.PaperlessBilling  \
0                          No         One year                      Yes   
1                         Yes   Month-to-month                       No   
2                          No   Month-to-month                      Yes   
3                         Yes   Month-to-month                    

# **Comprensión del DataSet**

In [187]:
# 1. Mostrar información básica del DataFrame
df_normalize.shape

(7267, 21)

In [188]:
# 2. Información detallada del DataFrame
df_normalize.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7267 entries, 0 to 7266
Data columns (total 21 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   customerID                 7267 non-null   object 
 1   Churn                      7267 non-null   object 
 2   customer.gender            7267 non-null   object 
 3   customer.SeniorCitizen     7267 non-null   int64  
 4   customer.Partner           7267 non-null   object 
 5   customer.Dependents        7267 non-null   object 
 6   customer.tenure            7267 non-null   int64  
 7   phone.PhoneService         7267 non-null   object 
 8   phone.MultipleLines        7267 non-null   object 
 9   internet.InternetService   7267 non-null   object 
 10  internet.OnlineSecurity    7267 non-null   object 
 11  internet.OnlineBackup      7267 non-null   object 
 12  internet.DeviceProtection  7267 non-null   object 
 13  internet.TechSupport       7267 non-null   objec

In [189]:
# 3. Explorar la columna target (Churn) - Variable objetivo
df_normalize['Churn'].value_counts() # Distribución de clientes que se fueron vs se quedaron

Churn
No     5174
Yes    1869
        224
Name: count, dtype: int64

In [190]:
# 3.1 Ver todos los valores únicos en Churn
df_normalize['Churn'].unique()

array(['No', 'Yes', ''], dtype=object)

Tenemos 224 registros "misteriosos":
Los valores únicos son: ['No', 'Yes', '']
¡Aquí tenemso un problema! Hay 224 registros con valores vacíos ('') en la columna Churn.

Lo que significa:

🟢 'No' = 5174 → Clientes que se quedaron

🔴 'Yes' = 1869 → Clientes que se fueron

⚠️ '' (vacío) = 224 → Registros sin información de Churn

In [191]:
# 3.2. Porcentaje de distribución de Churn
churn_porcentajes = df_normalize['Churn'].value_counts(normalize=True) * 100 # Porcentajes para entender el balance
churn_porcentajes.round(2)

Churn
No     71.20
Yes    25.72
        3.08
Name: proportion, dtype: float64

### Interpretación empresarial del resultado anterior:
🎯 Tasa de Churn = 25.72% es MUY ALTA para telecomunicaciones

📊 Retención = 71.20% podría mejorar significativamente

⚠️ 3.08% sin datos necesita limpieza

¿Qué significa esto?
**De cada 100 clientes, aproximadamente 26 se van cada período.**

Esta tasa de churn es preocupante. ***El objetivo será identificar qué factores llevan a ese 25.72% a irse***

## **🧹 Calidad de los datos**

In [192]:
# 1. Verifo, en el total de las columnas, si hay valores únicos en cada columna

for col in df_normalize.columns:
  print(f"valores únicos en la columna'{col}': {df_normalize[col].nunique()}")
  if df_normalize[col].nunique() < 20: # imprimir solo si son menos de 20 valores
    print(df_normalize[col].unique())
    print('-' * 50)  # Línea divisoria para mejor visualización de los datos

valores únicos en la columna'customerID': 7267
valores únicos en la columna'Churn': 3
['No' 'Yes' '']
--------------------------------------------------
valores únicos en la columna'customer.gender': 2
['Female' 'Male']
--------------------------------------------------
valores únicos en la columna'customer.SeniorCitizen': 2
[0 1]
--------------------------------------------------
valores únicos en la columna'customer.Partner': 2
['Yes' 'No']
--------------------------------------------------
valores únicos en la columna'customer.Dependents': 2
['Yes' 'No']
--------------------------------------------------
valores únicos en la columna'customer.tenure': 73
valores únicos en la columna'phone.PhoneService': 2
['Yes' 'No']
--------------------------------------------------
valores únicos en la columna'phone.MultipleLines': 3
['No' 'Yes' 'No phone service']
--------------------------------------------------
valores únicos en la columna'internet.InternetService': 3
['DSL' 'Fiber optic' 'No'

In [193]:
# 2. Verifico Números duplicados

duplicados = df_normalize.duplicated()
print("Numero de duplicados: ", duplicados.sum())

Numero de duplicados:  0


In [194]:
# 3. Verifico Valores nulos

valores_nulos = df_normalize.isnull().sum()

print(f' Valores nulos por columna:\n\n {valores_nulos}')

 Valores nulos por columna:

 customerID                   0
Churn                        0
customer.gender              0
customer.SeniorCitizen       0
customer.Partner             0
customer.Dependents          0
customer.tenure              0
phone.PhoneService           0
phone.MultipleLines          0
internet.InternetService     0
internet.OnlineSecurity      0
internet.OnlineBackup        0
internet.DeviceProtection    0
internet.TechSupport         0
internet.StreamingTV         0
internet.StreamingMovies     0
account.Contract             0
account.PaperlessBilling     0
account.PaymentMethod        0
account.Charges.Monthly      0
account.Charges.Total        0
dtype: int64


In [195]:
# 4.Verifico valores en blanco o vacios

valores_vacios = df_normalize.apply(lambda x: x.astype(str).str.strip() == '').sum()

print(f' Valores vacios por columna:\n\n {valores_vacios}')

 Valores vacios por columna:

 customerID                     0
Churn                        224
customer.gender                0
customer.SeniorCitizen         0
customer.Partner               0
customer.Dependents            0
customer.tenure                0
phone.PhoneService             0
phone.MultipleLines            0
internet.InternetService       0
internet.OnlineSecurity        0
internet.OnlineBackup          0
internet.DeviceProtection      0
internet.TechSupport           0
internet.StreamingTV           0
internet.StreamingMovies       0
account.Contract               0
account.PaperlessBilling       0
account.PaymentMethod          0
account.Charges.Monthly        0
account.Charges.Total         11
dtype: int64


# **🔧 Transformación**

In [196]:
# 1. Eliminando las columnas vacias de Churn

df_normalize = df_normalize[df_normalize['Churn'].str.strip() != '']
print("Número de filas después de eliminar las vacías en 'Churn':", len(df_normalize))

Número de filas después de eliminar las vacías en 'Churn': 7043


In [197]:
#2 . Cambiando la columna "account.Charges.Total" a Float

df_normalize['account.Charges.Total'] = pd.to_numeric(df_normalize['account.Charges.Total'], errors='coerce')
print(df_normalize['account.Charges.Total'].dtype)

float64


/tmp/ipython-input-3578694979.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_normalize['account.Charges.Total'] = pd.to_numeric(df_normalize['account.Charges.Total'], errors='coerce')


In [198]:
valores_vacios = df_normalize.apply(lambda x: x.astype(str).str.strip() == '').sum()

print(f' Valores vacios por columna:\n\n {valores_vacios}')

 Valores vacios por columna:

 customerID                   0
Churn                        0
customer.gender              0
customer.SeniorCitizen       0
customer.Partner             0
customer.Dependents          0
customer.tenure              0
phone.PhoneService           0
phone.MultipleLines          0
internet.InternetService     0
internet.OnlineSecurity      0
internet.OnlineBackup        0
internet.DeviceProtection    0
internet.TechSupport         0
internet.StreamingTV         0
internet.StreamingMovies     0
account.Contract             0
account.PaperlessBilling     0
account.PaymentMethod        0
account.Charges.Monthly      0
account.Charges.Total        0
dtype: int64


In [199]:
# 3. CREAR LA COLUMNA CUENTAS_DIARIAS
# Dividir la facturación mensual entre 30 días (promedio de días por mes)
df_normalize['Cuentas_Diarias'] = df_normalize['account.Charges.Monthly'] / 30
df_normalize['Cuentas_Diarias'] = df_normalize['Cuentas_Diarias'].round(2)


print(df_normalize[['account.Charges.Monthly', 'Cuentas_Diarias']])
df_normalize = df_normalize.drop('Cuentas_Diarias', errors='ignore')

      account.Charges.Monthly  Cuentas_Diarias
0                       65.60             2.19
1                       59.90             2.00
2                       73.90             2.46
3                       98.00             3.27
4                       83.90             2.80
...                       ...              ...
7262                    55.15             1.84
7263                    85.10             2.84
7264                    50.30             1.68
7265                    67.85             2.26
7266                    59.00             1.97

[7043 rows x 2 columns]


/tmp/ipython-input-3727456839.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_normalize['Cuentas_Diarias'] = df_normalize['account.Charges.Monthly'] / 30
/tmp/ipython-input-3727456839.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_normalize['Cuentas_Diarias'] = df_normalize['Cuentas_Diarias'].round(2)


In [200]:
# 4. VERIFICAR SI HAY VALORES NULOS EN LA NUEVA COLUMNA
df_normalize['Cuentas_Diarias'].isnull().sum()

np.int64(0)

In [201]:
df_normalize

customerID Churn customer.gender  customer.SeniorCitizen  \
0     0002-ORFBO    No          Female                       0   
1     0003-MKNFE    No            Male                       0   
2     0004-TLHLJ   Yes            Male                       0   
3     0011-IGKFF   Yes            Male                       1   
4     0013-EXCHZ   Yes          Female                       1   
...          ...   ...             ...                     ...   
7262  9987-LUTYD    No          Female                       0   
7263  9992-RRAMN   Yes            Male                       0   
7264  9992-UJOEL    No            Male                       0   
7265  9993-LHIEB    No            Male                       0   
7266  9995-HOTOH    No            Male                       0   

     customer.Partner customer.Dependents  customer.tenure phone.PhoneService  \
0                 Yes                 Yes                9                Yes   
1                  No                  No                9                Yes   
2                  No                  No                4                Yes   
3                 Yes                  No               13                Yes   
4                 Yes                  No                3                Yes   
...               ...                 ...              ...                ...   
7262               No                  No               13                Yes   
7263              Yes                  No               22                Yes   
7264               No                  No                2                Yes   
7265              Yes                 Yes               67                Yes   
7266              Yes                 Yes               63                 No   

     phone.MultipleLines internet.InternetService  ...  \
0                     No                      DSL  ...   
1                    Yes                      DSL  ...   
2                     No              Fiber optic  ...   
3                     No              Fiber optic  ...   
4                     No              Fiber optic  ...   
...                  ...                      ...  ...   
7262                  No                      DSL  ...   
7263                 Yes              Fiber optic  ...   
7264                  No                      DSL  ...   
7265                  No                      DSL  ...   
7266    No phone service                      DSL  ...   

     internet.DeviceProtection internet.TechSupport internet.StreamingTV  \
0                           No                  Yes                  Yes   
1                           No                   No                   No   
2                          Yes                   No                   No   
3                          Yes                   No                  Yes   
4                           No                  Yes                  Yes   
...                        ...                  ...                  ...   
7262                        No                  Yes                   No   
7263                        No                   No                   No   
7264                        No                   No                   No   
7265                       Yes                  Yes                   No   
7266                       Yes                   No                  Yes   

     internet.StreamingMovies account.Contract account.PaperlessBilling  \
0                          No         One year                      Yes   
1                         Yes   Month-to-month                       No   
2                          No   Month-to-month                      Yes   
3                         Yes   Month-to-month                      Yes   
4                          No   Month-to-month                      Yes   
...                       ...              ...                      ...   
7262                       No         One year                       No   
7263                      Yes 

In [202]:
df_normalize.describe()

customer.SeniorCitizen  customer.tenure  account.Charges.Monthly  \
count             7043.000000      7043.000000              7043.000000   
mean                 0.162147        32.371149                64.761692   
std                  0.368612        24.559481                30.090047   
min                  0.000000         0.000000                18.250000   
25%                  0.000000         9.000000                35.500000   
50%                  0.000000        29.000000                70.350000   
75%                  0.000000        55.000000                89.850000   
max                  1.000000        72.000000               118.750000   

       account.Charges.Total  Cuentas_Diarias  
count            7032.000000      7043.000000  
mean             2283.300441         2.158675  
std              2266.771362         1.003088  
min                18.800000         0.610000  
25%               401.450000         1.180000  
50%              1397.475000         2.340000  
75%              3794.737500         2.990000  
max              8684.800000         3.960000

## 5. Estandarización y transformación de datos


In [204]:
# 5.1. TRANSFORMACIÓN DE VARIABLES CATEGÓRICAS A BINARIAS
# Identificar columnas con valores Yes/No
yes_no_columns = []
for col in df_normalize.columns:
    if df_normalize[col].dtype == 'object':
        unique_vals = df_normalize[col].unique()
        if set(['Yes', 'No']).issubset(set(unique_vals)):
            yes_no_columns.append(col)

yes_no_columns

['Churn',
 'customer.Partner',
 'customer.Dependents',
 'phone.PhoneService',
 'phone.MultipleLines',
 'internet.OnlineSecurity',
 'internet.OnlineBackup',
 'internet.DeviceProtection',
 'internet.TechSupport',
 'internet.StreamingTV',
 'internet.StreamingMovies',
 'account.PaperlessBilling']

In [205]:
# 5.2. Verifico qué valores únicos tengo

print("VALORES ÚNICOS EN CADA COLUMNA: \n")

for col in yes_no_columns:
    if col in df_normalize.columns:
        unique_vals = df_normalize[col].unique()
        print(f"{col}: {unique_vals}")

VALORES ÚNICOS EN CADA COLUMNA: 

Churn: ['No' 'Yes']
customer.Partner: ['Yes' 'No']
customer.Dependents: ['Yes' 'No']
phone.PhoneService: ['Yes' 'No']
phone.MultipleLines: ['No' 'Yes' 'No phone service']
internet.OnlineSecurity: ['No' 'Yes' 'No internet service']
internet.OnlineBackup: ['Yes' 'No' 'No internet service']
internet.DeviceProtection: ['No' 'Yes' 'No internet service']
internet.TechSupport: ['Yes' 'No' 'No internet service']
internet.StreamingTV: ['Yes' 'No' 'No internet service']
internet.StreamingMovies: ['No' 'Yes' 'No internet service']
account.PaperlessBilling: ['Yes' 'No']


In [206]:
# 5.3. Mapeo explícito que maneja todos los casos
for col in yes_no_columns:
    if col in df_normalize.columns:
        # Mapeo que convierte 'No internet service' y 'No phone service' a 0
        df_normalize[col] = df_normalize[col].map({
            'Yes': 1,
            'No': 0,
            'No internet service': 0,  # Considerado como "No tiene el servicio" = 0
            'No phone service': 0      # Considerado como "No tiene el servicio" = 0
        })

        # Verificar si quedaron valores NaN (valores no mapeados)
        if df_normalize[col].isna().any():
            print(f"⚠️  ADVERTENCIA: {col} tiene valores no mapeados:")
            print(f"   Valores únicos: {df_normalize[col].value_counts(dropna=False)}")
        else:
            print(f"✅ {col}: Mapeo exitoso")

✅ Churn: Mapeo exitoso
✅ customer.Partner: Mapeo exitoso
✅ customer.Dependents: Mapeo exitoso
✅ phone.PhoneService: Mapeo exitoso
✅ phone.MultipleLines: Mapeo exitoso
✅ internet.OnlineSecurity: Mapeo exitoso
✅ internet.OnlineBackup: Mapeo exitoso
✅ internet.DeviceProtection: Mapeo exitoso
✅ internet.TechSupport: Mapeo exitoso
✅ internet.StreamingTV: Mapeo exitoso
✅ internet.StreamingMovies: Mapeo exitoso
✅ account.PaperlessBilling: Mapeo exitoso


In [207]:
# 5.4. RENOMBRAMIENTO DE COLUMNAS A ESPAÑOL
df_normalize.rename(columns={
    'customerID': 'Id_Cliente',
    'Churn': 'Fuga_Cliente',
    'customer.gender': 'Genero',
    'customer.SeniorCitizen': 'Ciudadano_Mayor',
    'customer.Partner': 'Tiene_Pareja',
    'customer.Dependents': 'Tiene_Dependientes',
    'customer.tenure': 'Antiguedad_Meses',
    'phone.PhoneService': 'Servicio_Telefono',
    'phone.MultipleLines': 'Lineas_Multiples',
    'internet.InternetService': 'Servicio_Internet',
    'internet.OnlineSecurity': 'Seguridad_Online',
    'internet.OnlineBackup': 'Respaldo_Online',
    'internet.DeviceProtection': 'Proteccion_Dispositivo',
    'internet.TechSupport': 'Soporte_Tecnico',
    'internet.StreamingTV': 'TV_Streaming',
    'internet.StreamingMovies': 'Peliculas_Streaming',
    'account.Contract': 'Tipo_Contrato',
    'account.PaperlessBilling': 'Facturacion_Digital',
    'account.PaymentMethod': 'Metodo_Pago',
    'account.Charges.Monthly': 'Cargo_Mensual',
    'account.Charges.Total': 'Cargo_Total',
    'Cuentas_Diarias': 'Cargo_Diario'
    }, inplace=True)

In [208]:
df_normalize.head()

Id_Cliente  Fuga_Cliente  Genero  Ciudadano_Mayor  Tiene_Pareja  \
0  0002-ORFBO             0  Female                0             1   
1  0003-MKNFE             0    Male                0             0   
2  0004-TLHLJ             1    Male                0             0   
3  0011-IGKFF             1    Male                1             1   
4  0013-EXCHZ             1  Female                1             1   

   Tiene_Dependientes  Antiguedad_Meses  Servicio_Telefono  Lineas_Multiples  \
0                   1                 9                  1                 0   
1                   0                 9                  1                 1   
2                   0                 4                  1                 0   
3                   0                13                  1                 0   
4                   0                 3                  1                 0   

  Servicio_Internet  ...  Proteccion_Dispositivo  Soporte_Tecnico  \
0               DSL  ...                       0                1   
1               DSL  ...                       0                0   
2       Fiber optic  ...                       1                0   
3       Fiber optic  ...                       1                0   
4       Fiber optic  ...                       0                1   

   TV_Streaming  Peliculas_Streaming   Tipo_Contrato  Facturacion_Digital  \
0             1                    0        One year                    1   
1             0                    1  Month-to-month                    0   
2             0                    0  Month-to-month                    1   
3             1                    1  Month-to-month                    1   
4             1                    0  Month-to-month                    1   

        Metodo_Pago  Cargo_Mensual Cargo_Total  Cargo_Diario  
0      Mailed check           65.6      593.30          2.19  
1      Mailed check           59.9      542.40          2.00  
2  Electronic check           73.9      280.85          2.46  
3  Electronic check           98.0     1237.85          3.27  
4      Mailed check           83.9      267.40          2.80  

[5 rows x 22 columns]

#**📊 Carga y análisis**

#**📄Informe final**